games-roguelike/nethack/nethack-3.6.0.ebuild

```diff
@@ -0,0 +1,186 @@
+		-e '/^WINTTYLIB/s| = .*| = '"$(
+				$(tc-getPKG_CONFIG) --libs ncurses
+			)"'|g' \
+		src/Makefile || die
+
+	if use X ; then
+		epatch "${FILESDIR}/${PV}-X-support.patch"
+	fi
+}
+
+src_compile() {
+	local lflags="${LDFLAGS}"
+
+	tc-export CC CXX LD AR RANLIB
+
+	cd "${S}"/src
+	append-flags -I../include
+
+	emake \
+		CC="$(tc-getCC)" \
+		CFLAGS="${CFLAGS}" \
+		LFLAGS="${lflags}" \
+		../util/makedefs \
+		|| die "initial makedefs build failed"
+	emake \
+		CC="$(tc-getCC)" \
+		CFLAGS="${CFLAGS}" \
+		LFLAGS="${lflags}" \
+		|| die "main build failed"
+	cd "${S}"/util
+	emake \
+		CC="$(tc-getCC)" \
+		CFLAGS="${CFLAGS}" \
+		LFLAGS="${lflags}" \
+		recover || die "util build failed"
+}
+
+src_install() {
+	emake \
+		CC="$(tc-getCC)" \
+		CFLAGS="${CFLAGS}" \
+		LFLAGS="-L/usr/X11R6/lib" \
+		GAMEPERM=0755 \
+		GAMEUID=root GAMEGRP=root \
+		PREFIX="${D}/usr" \
+		GAMEDIR="${D}${HACKDIR}" \
+		SHELLDIR="${D}/usr/bin" \
+		VARDIR="${D}"/var/lib/games/nethack \
+		install \
+		|| die "emake install failed"
+
+	rm -rf "${D}/usr/share/games/nethack/save"
+
+	newbin util/recover recover-nethack
+
+	# The final nethack is a sh script.  This fixes the hard-coded
+	# HACKDIR directory so it doesn't point to ${D}/usr/share/nethackdir
+	sed -i \
+		-e "s:^\(HACKDIR=\).*:\1${HACKDIR}:" \
+		"${D}/usr/bin/nethack" \
+		|| die "sed /usr/bin/nethack failed"
+
+	doman doc/*.6
+	dodoc doc/*.txt
+
+	# Can be copied to ~/.nethackrc to set options
+	# Add this to /etc/.skel as well, thats the place for default configs
+	insinto "${HACKDIR}"
+	doins "${FILESDIR}/dot.nethackrc"
+
+	local windowtypes="tty"
+	use X && windowtypes="${windowtypes} x11"
+	set -- ${windowtypes}
+	sed -i \
+		-e "s:GENTOO_WINDOWTYPES:${windowtypes}:" \
+		-e "s:GENTOO_DEFWINDOWTYPE:$1:" \
+		"${D}${HACKDIR}/dot.nethackrc" \
+		|| die "sed ${HACKDIR}/dot.nethackrc failed"
+	insinto /etc/skel
+	newins "${D}/${HACKDIR}/dot.nethackrc" .nethackrc
+
+	if use X ; then
+		# install nethack fonts
+		cd "${S}/win/X11"
+		bdftopcf -o nh10.pcf nh10.bdf || die "Converting fonts failed"
+		bdftopcf -o ibm.pcf ibm.bdf || die "Converting fonts failed"
+		insinto "${HACKDIR}/fonts"
+		doins *.pcf
+		cd "${D}/${HACKDIR}/fonts"
+		mkfontdir || die "The action mkfontdir ${HACKDIR}/fonts failed"
+
+		# copy nethack x application defaults
+		cd "${S}/win/X11"
+		insinto /etc/X11/app-defaults
+		newins NetHack.ad NetHack || die "Failed to install NetHack X app defaults"
+		sed -i \
+			-e 's:^!\(NetHack.tile_file.*\):\1:' \
+			"${D}/etc/X11/app-defaults/NetHack" \
+			|| die "sed /etc/X11/app-defaults/NetHack failed"
+	fi
+
+	local statedir="/var/lib/games/${PN}"
+	keepdir "${statedir}/save"
+	gamers_prep_vardata -R
+	gamers_preserve_vardata "${statedir}"/*
+	make_desktop_entry nethack "Nethack"
+	chmod o-r "${D}${statedir}" || die
+}
+
+pkg_postinst() {
+	elog "You may want to look at /etc/skel/.nethackrc for interesting options"
+	gamers_pkg_postinst
+}
+
+Status API Training Shop Blog About
```

Comment:
This is a copy of the 3.4.3 ebuild updated to 3.6.0.  Be aware that this package should be dropped from this overlay.  See https://packages.gentoo.org/packages/games-roguelike/nethack